[EWS] Introduce ValidateUserForQueue step

rkirsling opened the pull request:
#### f69f55a4ea5f56f03aa46ad4d9a4ba7d7963fb72
<pre>
[EWS] Introduce ValidateUserForQueue step
<a href="https://bugs.webkit.org/show_bug.cgi?id=283402">https://bugs.webkit.org/show_bug.cgi?id=283402</a>

Reviewed by Ryan Haddad.

* Tools/CISupport/ews-build/steps.py:
(ValidateUserForQueue):
(ValidateUserForQueue.__init__):
(ValidateUserForQueue.getResultSummary):
(ValidateUserForQueue.is_committer):
(ValidateUserForQueue.run):
* Tools/CISupport/ews-build/steps_unittest.py:

Canonical link: <a href="https://commits.webkit.org/287586@main">https://commits.webkit.org/287586@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/68818676db97fc4734ac6a9a52313fe587116b10

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/78968 "Passed style check") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/58005 "Build is in progress. Recent messages:OS: Sonoma (14.6), Xcode: 15.4; Skipping applying patch since patch_id isn't provided; Checked out pull request; Compiled WebKit (warnings)") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/32346 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/83609 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/30189 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/81101 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/67122 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/6278 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/61811 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/19734 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/82035 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/51850 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/71857 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/42115 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/49201 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/26040 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/28552 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/70315 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/26451 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/84994 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/6302 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/4349 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/70044 "Passed tests") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/78658 "Passed tests") | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/6466 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/67858 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/69296 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/17477 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/13343 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/12086 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/6252 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/6215 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/9666 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/8006 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rkirsling commented on Tools/CISupport/ews-build/config.json:
```diff
@@ -199,6 +199,11 @@
+    {
+      "name": "Validate-User-EWS", "shortname": "validate-user",
+      "factory": "ValidateUserFactory", "platform": "*",
+      "workernames": ["webkit-misc"]
```
(I left this value in because I'm still unclear on how or where this is meant to be deployed.)

briannafan commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -2039,6 +2039,55 @@ def send_email_for_github_failure(self):
+        else:
+            committer = self.getProperty('patch_committer', '').lower()
+
+        # For now, we assume that the requisite status is "committer".
```
nit: "For now" is a bit too vague, I'd either remove this line entirely or clarify with a `FIXME` if you intend on changing the required status after certain work is completed.

rkirsling replied:
I think we can remove it; I guess it was more of a note for reviewers than anything.

ryanhaddad reviewed (commented):
r=me

rkirsling committed:
[EWS] Introduce ValidateUserForQueue step
https://bugs.webkit.org/show_bug.cgi?id=283402

Reviewed by Ryan Haddad.

* Tools/CISupport/ews-build/steps.py:
(ValidateUserForQueue):
(ValidateUserForQueue.__init__):
(ValidateUserForQueue.getResultSummary):
(ValidateUserForQueue.is_committer):
(ValidateUserForQueue.run):
* Tools/CISupport/ews-build/steps_unittest.py:

Canonical link: https://commits.webkit.org/287586@main
